sdk/keyvault/azure.keyvault.keys.txt

```diff
@@ -0,0 +1,180 @@
+	def get_key(self, name: str, version: str=None, **kwargs) -> ~azure.keyvault.keys.models.Key: ...
+	def import_key(self, name: str, key: key, hsm: bool=None, enabled: bool=None, not_before: datetime.datetime=None, expires=None, tags: dict=None, **kwargs) -> ~azure.keyvault.keys.models.Key: ...
+	def list_deleted_keys(self, **kwargs) -> ~azure.core.paging.ItemPaged[~azure.keyvault.keys.models.DeletedKey]: ...
+	def list_key_versions(self, name: str, **kwargs) -> ~azure.core.paging.ItemPaged[~azure.keyvault.keys.models.KeyProperties]: ...
+	def list_keys(self, **kwargs) -> ~azure.core.paging.ItemPaged[~azure.keyvault.keys.models.KeyProperties]: ...
+	def purge_deleted_key(self, name: str, **kwargs): ...
+	def recover_deleted_key(self, name: str, **kwargs) -> ~azure.keyvault.keys.models.Key: ...
+	def restore_key(self, backup: bytes, **kwargs) -> ~azure.keyvault.keys.models.Key: ...
+	def update_key_properties(self, name: str, version: str=None, key_operations: key_operations=None, enabled: bool=None, expires: datetime.datetime=None, not_before: datetime.datetime=None, tags: dict=None, **kwargs) -> ~azure.keyvault.keys.models.Key: ...
+
+class KeyCurveName(str, Enum, object):
+	...
+
+class KeyOperation(str, Enum, object):
+	...
+
+class KeyType(str, Enum, object):
+	...
+
+**** Package/Module azure.keyvault.keys.models****
+class DeletedKey(Key, object):
+	deleted_date
+	id
+	key_material
+	name
+	properties
+	recovery_id
+	scheduled_purge_date
+
+	def __init__(self, properties, key_material=None, deleted_date=None, recovery_id=None, scheduled_purge_date=None): ...
+
+class JsonWebKey(object):
+	kid:str
+	kty:kty
+	key_ops:key_ops
+	n:bytes
+	e:bytes
+	d:bytes
+	dp:bytes
+	dq:bytes
+	qi:bytes
+	p:bytes
+	q:bytes
+	k:bytes
+	t:bytes
+	crv:crv
+	x:bytes
+	y:bytes
+
+	def __init__(self, *, kid, kty, key_ops, n, e, d, dp, dq, qi, p, q, k, t, crv, x, y, **kwargs): ...
+
+class Key(object):
+	id
+	key_material
+	name
+	properties
+
+	def __init__(self, properties, key_material): ...
+
+Warning - couldn't get source for __new__
+class KeyOperationResult(tuple, object):
+	id
+	value
+
+	def __getnewargs__(self): ...
+	def __new__(_cls, id, value): ...
+	def __repr__(self): ...
+
+class KeyProperties(object):
+	created
+	enabled
+	expires
+	id
+	managed
+	name
+	not_before
+	recovery_level
+	tags
+	updated
+	vault_url
+	version
+
+	def __init__(self, attributes, vault_id, **kwargs): ...
+
+**** Package/Module azure.keyvault.keys.crypto****
+class CryptographyClient(KeyVaultClientBase, object):
+	key:key
+	credential
+
+	key_id
+	vault_url
+
+	def __init__(self, key: key, credential, **kwargs): ...
+	def decrypt(self, algorithm: algorithm, ciphertext: bytes, **kwargs) -> :class:`~azure.keyvault.keys.crypto.DecryptResult`: ...
+	def encrypt(self, algorithm: algorithm, plaintext: bytes, **kwargs) -> :class:`~azure.keyvault.keys.crypto.EncryptResult`: ...
+	def get_key(self, **kwargs) -> :class:`~azure.keyvault.keys.models.Key` or ``None``: ...
```

Comment:
Being you need a key to create this instance, I'm confused why this guy could return None?

Reply:
It reflects the Key Vault permissions model. A user may be permitted to perform (service-side) operations with a key but not to get that key. So, the client can be initialized with a `Key` instance or a key's identifier (`str`). When an identifier is used and the client lacks permission to get the key, this method returns `None`.

Reply:
Agree to fix. #7784 is tracking this issue